collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -704,13 +701,19 @@ public void run() {
                 break;
             case AUDIO_CAPTURE:
             case VIDEO_CAPTURE:
-                Uri mediaUri = intent.getData();
-                saveAudioVideoAnswer(mediaUri);
-                String filePath = MediaUtils.getDataColumn(this, mediaUri, null, null);
-                if (filePath != null) {
-                    new File(filePath).delete();
+                try{
```

Comment:
In this case when request code  == REQUESTED_CANCELLED, it may have chance that user has accidently pressed back button, or some other reason. So asking him to try again would be a good thing.